testnet/stacks-node/src/neon_node.rs

```diff
@@ -979,10 +1003,98 @@ fn spawn_miner_relayer(
                                         "Microblock miner tip is now {}/{} ({})",
                                         &consensus_hash, &block_header_hash, StacksBlockHeader::make_index_block_hash(&consensus_hash, &block_header_hash)
                                     );
+
                                     miner_tip = Some((ch, bh, microblock_privkey));
+                                    let parent_index_hash = StacksBlockHeader::make_index_block_hash(&ch, &bh);
+                                    let cost_so_far = StacksChainState::get_stacks_block_anchored_cost(
+                                        chainstate.db(),
+                                        &parent_index_hash,
+                                    ).expect("BUG: failed to load anchored cost of parent block")
+                                    .ok_or(NetError::NotFoundError).expect("BUG: failed to load anchored cost of parent block");
+
+                                    // Before updating the miner state, we need to grab the unconfirmed microblocks from the parent
+                                    // block, to be packaged into a microblock for this new anchor block.
+                                    let prev_microblocks = if let Some(miner_state) = microblock_miner_state {
+                                        miner_state.unconfirmed_microblocks
+                                    } else {
+                                        vec![]
+                                    };
```

Comment:
There's a simpler way to write this:

```suggestion
                                    let prev_microblocks = microblock_miner_state
                                        .map(|state| state.unconfirmed_microblocks)
                                        .unwrap_or_default();
```